### PR TITLE
Feature/12 to design form for sign up possibility

### PR DESCRIPTION
**Pull request by SergiyKozachok**

### CODE REVIEW

File: InternetHospitalWebUI/src/app/Components/Authorization/sign-up/sign-up.component.ts [hunk 1]
```diff
@@ -7,9 +10,21 @@ import { Component, OnInit } from '@angular/core';
...
   }
 
+  onClear() {
+    this.service.form.reset();
+    this.service.initializeFormGroup();
+  }
+
+  onSubmit(form: NgForm) {
+    if(this.service.form.valid) {
+      this.service.postUser(form.value).subscribe(res => console.log(res));
```
Comment (andriipetryna) [inline InternetHospitalWebUI/src/app/Components/Authorization/sign-up/sign-up.component.ts]: The result must be shown for a user in some way. If the registration failed - the user must see an error message.
Output to the console is redundant, remove it.

File: InternetHospitalWebUI/src/app/Services/authorization.service.ts [hunk 2]
```diff
@@ -1,9 +1,44 @@
...
+      ConfirmPassword: '',
+      Role: '',
+      AddressFull: '',
+    });
+  }
 
-  constructor() { }
+  postUser(user: User) {    
+    var body = JSON.stringify(user);
+    console.log(body);
```
Comment (andriipetryna) [inline InternetHospitalWebUI/src/app/Services/authorization.service.ts]: Output to the console is redundant, remove it.

File: InternetHospitalWebUI/src/app/Models/User.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export class User {
+    Id: number;
+    UserName: string;
+    Email: string;
+    Password: string;
+    ConfirmPassword: string;
```
Comment (andriipetryna) [inline InternetHospitalWebUI/src/app/Models/User.ts:6]: Confirm password field is unnecessary on the server side. If password and confirm password fields don't match - you can simply make registration button inactive in front-end. So if a user could send a registration data to the server it means his password fields match, and you will always get two similar passwords on the server side.

File: InternetHospitalWebUI/src/app/config.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export const configUrl = 'https://localhost:44390';
```
Comment (andriipetryna) [inline InternetHospitalWebUI/src/app/config.ts]: Maybe it's better to rename configUrl variable to HOST_URL or ROOT_URL? 
And name constants in uppercase with underscore between words, please.

File: InternetHospitalWebUI/src/app/Modules/routing/routing.module.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  imports: [
+    CommonModule,
+    RouterModule.forRoot(routes)
+  ],
+  declarations: [],
+  exports: [RouterModule]
+})
+export class RoutingModule { }
+
+export const routingComponents = [
```
Comment (andriipetryna) [inline InternetHospitalWebUI/src/app/Modules/routing/routing.module.ts]: Rename constant to uppercase

File: InternetHospitalWebUI/src/app/Components/Authorization/sign-up/sign-up.component.ts [hunk 6]
```diff
@@ -7,9 +10,21 @@ import { Component, OnInit } from '@angular/core';
...
-  constructor() { }
+  constructor(private service: AuthorizationService) { }
 
   ngOnInit() {
   }
 
+  onClear() {
+    this.service.form.reset();
+    this.service.initializeFormGroup();
+  }
```
Comment (andriipetryna) [inline InternetHospitalWebUI/src/app/Components/Authorization/sign-up/sign-up.component.ts:20]: What the difference between `form.reset()` and `service.initializeFormGroup()` method where you simply assign empty values for form's inputs? Does not `form.reset()` make the same action?

**Review by andriipetryna — CHANGES_REQUESTED** [review]

**Comment by eugent482** [comment]
I think you should rename authorization.service to registration.service. Authorization is more about sign in.

File: InternetHospitalWebUI/src/app/Components/Authorization/sign-up/sign-up.component.ts (hunk 1 above)
Comment (SergiyKozachok) [reply InternetHospitalWebUI/src/app/Components/Authorization/sign-up/sign-up.component.ts]: Yes, we will create a nitificationService, and there will be a single format of notifications. But this is a separate task, and we will be obliged to do it in the next sprint

File: InternetHospitalWebUI/src/app/Models/User.ts (hunk 3 above)
Comment (SergiyKozachok) [reply InternetHospitalWebUI/src/app/Models/User.ts:6]: I understand this, but Edik wants to do one more validation on the server, and he needs this field

**Comment by SergiyKozachok** [comment]
> I think you should rename authorization.service to registration.service. Authorization is more about sign in.

I understand this, but it seems to me that on Sign Up and Sign In we can use one service. It is not necessary for each component to create a separate service

**Review by andriipetryna — APPROVED** [review]

**Review by serhiidotsyk — APPROVED** [review]

**Review by TREN-D — APPROVED** [review]

**Pull request merged**